### PR TITLE
metallbv2-add-connect-timer

### PR DESCRIPTION
**Pull request by gkopels**
Adding metallb connect timer. Connect timer is part of mlbtypesv1beta2

### CODE REVIEW

File: pkg/metallb/bgppeer.go [hunk 1]
```diff
@@ -377,6 +378,29 @@ func (builder *BGPPeerBuilder) WithKeepalive(keepalive metav1.Duration) *BGPPeer
...
 }
 
+// WithConnectTimer defines the reconnect timer between BGP neighbors.
+func (builder *BGPPeerBuilder) WithConnectTimer(connectTimer metav1.Duration) *BGPPeerBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating BGPPeer %s in namespace %s with this holdTime: %s",
```
Comment (evgenLevin) [inline pkg/metallb/bgppeer.go:388]: change holdTime to  ConnectTime

File: pkg/metallb/bgppeer.go [hunk 2]
```diff
@@ -377,6 +378,29 @@ func (builder *BGPPeerBuilder) WithKeepalive(keepalive metav1.Duration) *BGPPeer
...
+		builder.Definition.Name, builder.Definition.Namespace, connectTimer)
+
+	duration := connectTimer.Duration
+
+	if duration < time.Second || duration > 65535*time.Second {
+		glog.V(100).Infof("A valid connect time is between 1-65535")
+
+		builder.errorMsg = "bgppeer 'connectTime' value is not valid"
+	}
+
```
Comment (evgenLevin) [inline pkg/metallb/bgppeer.go:398]: please add

	if builder.errorMsg != "" {
		return builder
	}

File: pkg/metallb/bgppeer.go [hunk 3]
```diff
@@ -397,8 +421,9 @@ func (builder *BGPPeerBuilder) WithNodeSelector(nodeSelector map[string]string)
 		return builder
 	}
 
-	ndSelector := []mlbtypes.NodeSelector{{MatchLabels: nodeSelector}}
-	builder.Definition.Spec.NodeSelectors = ndSelector
+	builder.Definition.Spec.NodeSelectors = []metav1.LabelSelector{{
```
Comment (evgenLevin) [inline pkg/metallb/bgppeer.go:424]: What's the idea of this change?

File: pkg/metallb/bgppeer_test.go [hunk 4]
```diff
@@ -291,11 +329,6 @@ func TestBGPPeerWithNodeSelector(t *testing.T) {
 			testBGPPeer:  buildValidBGPPeerBuilder(buildBGPPeerTestClientWithDummyObject()),
 			nodeSelector: map[string]string{"test": "test1"},
 		},
-		{
-			testBGPPeer:   buildInValidBGPPeerBuilder(buildBGPPeerTestClientWithDummyObject()),
```
Comment (evgenLevin) [inline pkg/metallb/bgppeer_test.go:295]: What's the idea of this change?

**Review by evgenLevin — COMMENTED** [review]
several comments

File: pkg/metallb/bgppeer.go (hunk 3 above)
Comment (gkopels) [reply pkg/metallb/bgppeer.go:424]: They changed it upstream in mlbtypesv1beta2

File: pkg/metallb/bgppeer.go (hunk 2 above)
Comment (gkopels) [reply pkg/metallb/bgppeer.go:398]: added

File: pkg/metallb/bgppeer_test.go (hunk 4 above)
Comment (gkopels) [reply pkg/metallb/bgppeer_test.go]: replaced

File: pkg/metallb/bgppeer.go (hunk 1 above)
Comment (gkopels) [reply pkg/metallb/bgppeer.go]: changed

File: pkg/metallb/bgppeer.go [hunk 5]
```diff
@@ -377,12 +378,43 @@ func (builder *BGPPeerBuilder) WithKeepalive(keepalive metav1.Duration) *BGPPeer
...
+	return builder
+}
+
 // WithNodeSelector defines the nodeSelector placed in the BGPPeer spec.
 func (builder *BGPPeerBuilder) WithNodeSelector(nodeSelector map[string]string) *BGPPeerBuilder {
 	if valid, _ := builder.validate(); !valid {
 		return builder
 	}
 
+	if builder.errorMsg != "" {
```
Comment (evgenLevin) [inline pkg/metallb/bgppeer.go:414]: ?

**Review by evgenLevin — COMMENTED** [review]
1 comment

File: pkg/metallb/bgppeer.go (hunk 5 above)
Comment (gkopels) [reply pkg/metallb/bgppeer.go]: removed

**Review by klaskosk — APPROVED** [review]
/lgtm

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**